vpr/src/base/region.h

```diff
@@ -4,6 +4,40 @@
 #include <vtr_geometry.h>
 #include "vpr_types.h"
 
+struct RegionRectCoord {
```

Comment:
Make a nice doxygen comment on what this datastructure is and what it is used for. Comment each member variable briefly.